unittests/src/logging-tests.cpp

```diff
@@ -0,0 +1,137 @@
+        logger.info("Test log");
+        REQUIRE_FALSE(logger.context_info().contains("key1"));
+        REQUIRE(logger.context_info().contains("key2"));
+        REQUIRE(dummy_sink_->last_log().context_info.contains("key2"));
+        REQUIRE_FALSE(dummy_sink_->last_log().context_info.contains("key1"));
+    }
+
+    SECTION("Clear Context Info")
+    {
+        ContextInfo ci{{"key1", "value1"}, {"key2", "value2"}};
+        logger.add_context_keys(ci);
+        logger.clear_context_info();
+        logger.info("Test log");
+        REQUIRE(logger.context_info().empty());
+        REQUIRE(dummy_sink_->last_log().context_info.empty());
+    }   
+}
+
+TEST_CASE_METHOD(LoggingTestsFixture, "Logger Manager Global Context Info Tests", "[logger]")
+{
+    auto& manager = octo::logger::Manager::instance();
+
+    SECTION("Init")
+    {
+        REQUIRE(manager.global_context_info().empty());
+    }
+
+    SECTION("Update Global Context Info")
+    {
+        ContextInfo ci{{"key1", "value1"}, {"key2", "value2"}};
+        manager.set_global_context_info(std::move(ci));
+        Logger logger("logging-tests");
+        logger.info("Test log");
+        REQUIRE_FALSE(manager.global_context_info().empty());
+        REQUIRE(manager.global_context_info().contains("key1"));
+        REQUIRE(manager.global_context_info().contains("key2"));
+        REQUIRE(dummy_sink_->last_log().context_info == ci);
+    }
+
+    SECTION("With Logger Context Info")
+    {
+        ContextInfo ci{{"key1", "value1"}, {"key2", "value2"}};
+        manager.set_global_context_info(std::move(ci));
+        Logger logger("logging-tests");
+        logger.add_context_keys({{"key3", "value3"}});
+        logger.info("Test log");
+        REQUIRE(dummy_sink_->last_log().context_info.contains("key1"));
+        REQUIRE(dummy_sink_->last_log().context_info.contains("key2"));
+        REQUIRE(dummy_sink_->last_log().context_info.contains("key3"));
+    }
+
+    SECTION("With Log Context Info")
+    {
+        ContextInfo ci{{"key1", "value1"}, {"key2", "value2"}};
+        manager.set_global_context_info(std::move(ci));
+        Logger logger("logging-tests");
+        logger.add_context_keys({{"key3", "value3"}, {"key4", "value4"}});
+        logger.info("Test log").with_context({{"key5", "value5"}});
+        REQUIRE(dummy_sink_->last_log().context_info.contains("key1"));
+        REQUIRE(dummy_sink_->last_log().context_info.contains("key2"));
+        REQUIRE(dummy_sink_->last_log().context_info.contains("key3"));
+        REQUIRE(dummy_sink_->last_log().context_info.contains("key4"));
+        REQUIRE(dummy_sink_->last_log().context_info.contains("key5"));
+    }
+
+    SECTION("With New Context Info")
+    {
+        ContextInfo ci{{"key5", "value5"}};
+        manager.set_global_context_info(std::move(ci));
+        Logger logger("logging-tests");
+        logger.add_context_keys({{"key3", "value3"}});
+        logger.info("Test log").with_context({{"key4", "value4"}});
```

Comment:
Missing test that using the same logger again would not contain the ContextInfo passed in `with_context()`.

Reply:
added